service/src/test/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceTest.java

```diff
@@ -322,57 +334,142 @@ private BaseResource getResource() {
     @DataProvider(name = "filterPolicySetsInvalidRequestDataProvider")
     private Object[][] filterPolicySetsInvalidRequestDataProvider()
             throws JsonParseException, JsonMappingException, IOException {
-        return new Object[][] {
-                { createOnePolicySet(), Arrays.asList(new String[] { "Invalid policy set id" }) },
-                { createTwoPolicySets(), Arrays.asList(new String[] {}) },
-                { createTwoPolicySets(), Arrays
-                        .asList(new String[] { "policy-set-with-one-policy-one-condition", "noexistent-policy-set" }) } };
+        List<PolicySet> onePolicySet = createOnePolicySet();
+        List<PolicySet> twoPolicySets = createTwoPolicySets();
+        return new Object[][] { filterOnePolicySetByNonexistentPolicySet(onePolicySet),
+                filterTwoPolicySetsByNull(twoPolicySets), filterTwoPolisySetsByEmptyList(twoPolicySets),
+                filterTwoPolicySetsByByNonexistentPolicySet(twoPolicySets) };
+    }
+
+    private Object[] filterOnePolicySetByNonexistentPolicySet(List<PolicySet> onePolicySet) {
+        return new Object[] { onePolicySet, Arrays.asList(new String[] { "nonexistent-policy-set" }) };
+    }
+
+    private Object[] filterTwoPolicySetsByNull(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, null };
+    }
+
+    private Object[] filterTwoPolisySetsByEmptyList(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, Collections.emptyList() };
+    }
+
+    private Object[] filterTwoPolicySetsByByNonexistentPolicySet(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, Arrays
+                .asList(new String[] { twoPolicySets.get(0).getName(), "noexistent-policy-set" }) };
     }
 
     @DataProvider(name = "filterPolicySetsDataProvider")
-    private Object[][] filterPolicySetsDataProvider() throws JsonParseException, JsonMappingException, IOException {
+    private Object[][] filterPolicySetsDataProvider() {
         List<PolicySet> onePolicySet = createOnePolicySet();
         List<PolicySet> twoPolicySets = createTwoPolicySets();
-        return new Object[][] {
-            { onePolicySet, null, onePolicySet },
-            { onePolicySet, Collections.emptyList(), onePolicySet },
-            { onePolicySet, Arrays.asList(onePolicySet.get(0).getName()), onePolicySet },
-            { twoPolicySets, Arrays.asList(twoPolicySets.get(0).getName()), twoPolicySets.subList(0, 1) },
-            { twoPolicySets, Arrays.asList(twoPolicySets.get(1).getName()), twoPolicySets.subList(1, 2) },
-            { twoPolicySets,
-                Arrays.asList(twoPolicySets.get(0).getName(), twoPolicySets.get(1).getName()), twoPolicySets }
+        return new Object[][] { filterNullPolicySetsListByNull(),
```

Comment:
@sanjeevchopra,
Removing filterNullPolicySetsListByNull() and filterEmptyPolicySetsListByNull().  Implementation of filterPolicySetsByPriority() relies on the "contract", evalPolicy(), to check that policy sets are not null or empty.  Tests for evaluation order being null or empty, like filterOnePolicySetByNull() and filterOnePolicySetByEmptyList, are still needed.